tests/lib/collectors/yara.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+"""Tests the Yeti YaraCollector collectors (YetiYaraCollector)."""
+
+from unittest import mock
+import unittest
+
+from dftimewolf.lib import state
+from dftimewolf.lib.collectors import yara
+from dftimewolf.lib.containers import containers
+
+from dftimewolf import config
+
+MOCK_RESPONSE = [{
+    'created': '2022-11-18T10:54:12.039366Z',
+    'description': 'rule description',
+    'id': 'x-yara--2a2c4c6c-1797-4add-9787-19890ac35fc3',
+    'labels': ['label1', 'label2'],
+    'modified': '2022-11-18T10:58:36.970528Z',
+    'name': 'Some random rule',
+    'pattern': 'rule test {\n'
+                '  meta:\n'
+                '    author = "tomchop"\n'
+                '  strings:\n'
+                '    $s = "r4nd0m"\n'
+                '  condition:\n'
+                '    any of them\n'
+                '}',
+    'spec_version': '2.1',
+    'type': 'x-yara',
+    'valid_from': '2022-11-17T23:00:00Z'
+}]
+
+class YetiYaraCollectorTest(unittest.TestCase):
```

Comment:
I reckon it might be worth adding in a failure case, that is, if the server response is something other than `200 OK`.